HeavyIonsAnalysis/JetAnalysis/src/HiInclusiveJetAnalyzer.cc

```diff
@@ -1464,6 +1481,25 @@ HiInclusiveJetAnalyzer::analyze(const Event& iEvent,
 	azimuth_adapt = azimuth_adapt_new;
       }
     }
+
+    //recluster the jet constituents in reWTA scheme-------------------------
+    // disable for calo jets
+    if(doWTARecluster_ && !(jetName_.find("Calo") != std::string::npos)){
+      std::vector<fastjet::PseudoJet> candidates;
+      auto daughters = jet.getJetConstituents();
+      for(auto it = daughters.begin(); it!=daughters.end(); ++it){
+        candidates.push_back(fastjet::PseudoJet((**it).px(), (**it).py(), (**it).pz(), (**it).energy()));
+      }
+      auto cs = new fastjet::ClusterSequence(candidates, WTAjtDef);
+      std::vector<fastjet::PseudoJet> wtajt = fastjet::sorted_by_pt(cs->inclusive_jets(0));
+
+      jets_.WTAeta[jets_.nref] = wtajt[0].eta();
```

Comment:
I would add protection here against an empty vector.  If for some reason the list of constituents from the PF jet is empty, the code will crash since WTA jet collection is empty.

Something like
jets_.WTAeta[jets_.nref] = (wtajt.size() > 0) ? wtajt[0].eta() : -999;